File: Raven.Abstractions/Data/DocumentsChanges.cs
using System;
namespace Raven.Abstractions.Data
{
    public class DocumentsChanges
    {
        /// <summary>
        /// Previous field value.
        /// </summary>
        public string FieldOldValue { get; set; }

        /// <summary>
        /// Current field value.
        /// </summary>
        public string FieldNewValue { get; set; }

        /// <summary>
        /// Previous field type.
        /// </summary>
        public string FieldOldType { get; set; }

        /// <summary>
        /// Current field type.
        /// </summary>
        public string FieldNewType { get; set; }

        /// <summary>
        /// Type of change that occured.
        /// </summary>
        public ChangeType Change { get; set; }

        /// <summary>
        /// Name of field on which the change occured.
        /// </summary>
        public string FieldName { get; set; }

        public enum ChangeType
        {
            DocumentDeleted,
            DocumentAdded,
            FieldChanged,
            NewField,
            RemovedField,
            ArrayValueAdded,
            ArrayValueRemoved
        }

        protected bool Equals(DocumentsChanges other)
        {
            return string.Equals(FieldOldValue, other.FieldOldValue)
                   && string.Equals(FieldNewValue, other.FieldNewValue)
                   && string.Equals(FieldOldType, other.FieldOldType)
                   && string.Equals(FieldName, other.FieldName)
                   && string.Equals(FieldNewType, other.FieldNewType)
                   && Change ==  other.Change;
        }

        public override int GetHashCode()
        {
            unchecked
            {
                int hashCode = (FieldOldValue != null ? FieldOldValue.GetHashCode() : 0);
                hashCode = (hashCode*397) ^ (FieldNewValue != null ? FieldNewValue.GetHashCode() : 0);
                hashCode = (hashCode*397) ^ (FieldOldType != null ? FieldOldType.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (FieldNewType != null ? FieldNewType.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (FieldName != null ? FieldName.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (Change.GetHashCode());
                return hashCode;
            }
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != GetType()) return false;
            return Equals((DocumentsChanges) obj);
        }
    }
    
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/data/DocumentsChanges.java
package net.ravendb.abstractions.data;


public class DocumentsChanges {

  private String fieldOldValue;
  private String fieldNewValue;
  private String fieldOldType;
  private String fieldNewType;
  private ChangeType change;
  private String fieldName;

  public String getFieldOldValue() {
    return fieldOldValue;
  }

  public void setFieldOldValue(String fieldOldValue) {
    this.fieldOldValue = fieldOldValue;
  }

  public String getFieldNewValue() {
    return fieldNewValue;
  }

  public void setFieldNewValue(String fieldNewValue) {
    this.fieldNewValue = fieldNewValue;
  }

  public String getFieldOldType() {
    return fieldOldType;
  }

  public void setFieldOldType(String fieldOldType) {
    this.fieldOldType = fieldOldType;
  }

  public String getFieldNewType() {
    return fieldNewType;
  }

  public void setFieldNewType(String fieldNewType) {
    this.fieldNewType = fieldNewType;
  }

  public ChangeType getChange() {
    return change;
  }

  public void setChange(ChangeType change) {
    this.change = change;
  }

  public String getFieldName() {
    return fieldName;
  }

  public void setFieldName(String fieldName) {
    this.fieldName = fieldName;
  }

  public static enum ChangeType {
    DOCUMENT_DELETED,
    DOCUMENT_ADDED,
    FIELD_CHANGED,
    NEW_FIELD,
    REMOVED_FIELD,
    ARRAY_VALUE_ADDED,
    ARRAY_VALUE_REMOVED;
  }

  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + ((change == null) ? 0 : change.hashCode());
    result = prime * result + ((fieldName == null) ? 0 : fieldName.hashCode());
    result = prime * result + ((fieldNewType == null) ? 0 : fieldNewType.hashCode());
    result = prime * result + ((fieldNewValue == null) ? 0 : fieldNewValue.hashCode());
    result = prime * result + ((fieldOldType == null) ? 0 : fieldOldType.hashCode());
    result = prime * result + ((fieldOldValue == null) ? 0 : fieldOldValue.hashCode());
    return result;
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) return true;
    if (obj == null) return false;
    if (getClass() != obj.getClass()) return false;
    DocumentsChanges other = (DocumentsChanges) obj;
    if (change != other.change) return false;
    if (fieldName == null) {
      if (other.fieldName != null) return false;
    } else if (!fieldName.equals(other.fieldName)) return false;
    if (fieldNewType == null) {
      if (other.fieldNewType != null) return false;
    } else if (!fieldNewType.equals(other.fieldNewType)) return false;
    if (fieldNewValue == null) {
      if (other.fieldNewValue != null) return false;
    } else if (!fieldNewValue.equals(other.fieldNewValue)) return false;
    if (fieldOldType == null) {
      if (other.fieldOldType != null) return false;
    } else if (!fieldOldType.equals(other.fieldOldType)) return false;
    if (fieldOldValue == null) {
      if (other.fieldOldValue != null) return false;
    } else if (!fieldOldValue.equals(other.fieldOldValue)) return false;
    return true;
  }



}
